coderd/taskname/taskname.go

```diff
@@ -79,33 +105,82 @@ func generateSuffix() string {
 	return fmt.Sprintf("%04x", num)
 }
 
-func GenerateFallback() string {
+// generateFallback generates a random task name when other methods fail.
+// Uses Docker-style name generation with a collision-resistant suffix.
+func generateFallback() TaskName {
 	// We have a 32 character limit for the name.
-	// We have a 5 character prefix `task-`.
 	// We have a 5 character suffix `-ffff`.
-	// This leaves us with 22 characters for the middle.
+	// This leaves us with 27 characters for the name.
 	//
 	// Unfortunately, `namesgenerator.GetRandomName(0)` will
-	// generate names that are longer than 22 characters, so
+	// generate names that are longer than 27 characters, so
 	// we just trim these down to length.
 	name := strings.ReplaceAll(namesgenerator.GetRandomName(0), "_", "-")
-	name = name[:min(len(name), 22)]
+	name = name[:min(len(name), 27)]
 	name = strings.TrimSuffix(name, "-")
 
-	return fmt.Sprintf("task-%s-%s", name, generateSuffix())
+	taskName := fmt.Sprintf("%s-%s", name, generateSuffix())
+	displayName := strings.ReplaceAll(name, "-", " ")
+	if len(displayName) > 0 {
+		displayName = strings.ToUpper(displayName[:1]) + displayName[1:]
+	}
+
+	return TaskName{
+		Name:        taskName,
+		DisplayName: displayName,
+	}
 }
 
-func Generate(ctx context.Context, prompt string, opts ...Option) (string, error) {
-	o := options{}
-	for _, opt := range opts {
-		opt(&o)
+// generateFromPrompt creates a task name directly from the prompt by sanitizing it.
+// This is used as a fallback when Claude fails to generate a name.
+func generateFromPrompt(prompt string) (TaskName, error) {
+	// Normalize newlines and tabs to spaces
+	prompt = regexp.MustCompile(`[\n\r\t]+`).ReplaceAllString(prompt, " ")
+
+	// Truncate prompt to 27 chars with full words for task name generation
+	truncatedForName := prompt
+	if len(prompt) > 27 {
+		truncatedForName = strutil.Truncate(prompt, 27, strutil.TruncateWithFullWords)
 	}
 
-	if o.model == "" {
-		o.model = defaultModel
+	// Generate task name from truncated prompt
+	name := strings.ToLower(truncatedForName)
+	name = strings.ReplaceAll(name, " ", "-")
+	name = regexp.MustCompile(`[^a-z0-9-]+`).ReplaceAllString(name, "")
```

Comment:
See motivation above and below.
```suggestion
	name = regexp.MustCompile(`[^a-z0-9]`).ReplaceAllString(name, "")
```